scipy/stats/tests/test_sensitivity_analysis.py

```diff
@@ -0,0 +1,205 @@
+import numpy as np
+from numpy.testing import assert_allclose
+import pytest
+from scipy.stats._resampling import BootstrapResult
+
+from scipy.stats import f_ishigami, sobol_indices, uniform
+from scipy.stats._sensitivity_analysis import sample_AB, sample_A_B
+
+@pytest.fixture(scope='session')
+def ishigami_ref_indices():
+    """Reference values for Ishigami from Saltelli2007.
+
+    Chapter 4, exercise 5 pages 179-182.
+    """
+    a = 7.
+    b = 0.1
+
+    var = 0.5 + a ** 2 / 8 + b * np.pi ** 4 / 5\
+        + b ** 2 * np.pi ** 8 / 18
+    v1 = 0.5 + b * np.pi ** 4 / 5 + b ** 2 * np.pi ** 8 / 50
+    v2 = a ** 2 / 8
+    v3 = 0
+    v12 = 0
+    v13 = b ** 2 * np.pi ** 8 * 8 / 225
+    v23 = 0
+
+    s_first = np.array([v1 / var, v2 / var, v3 / var])
+    s_second = np.array([
+        [0., 0., v13 / var],
+        [v12 / var, 0., v23 / var],
+        [v13 / var, v23 / var, 0.]
+    ])
+    s_total = s_first + s_second.sum(axis=1)
+
+    return [s_first.reshape(1, -1), s_total.reshape(1, -1)]
+
+
+def f_ishigami_vec(x):
+    res = f_ishigami(x)
+    return np.column_stack([res, res])
```

Comment:
Is the idea that the output of a function can be ND? It says "_vec" which suggests "vectorized" to me. What is the use case?

Reply:
Yes this is to simulate having 2 outputs, so (n, s) here (n, 2)

Reply:
Oops I confused myself. Got it.